engine/src/main/java/io/camunda/zeebe/engine/state/appliers/ProcessInstanceCreationCreatedApplier.java

```diff
@@ -0,0 +1,107 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH under
+ * one or more contributor license agreements. See the NOTICE file distributed
+ * with this work for additional information regarding copyright ownership.
+ * Licensed under the Zeebe Community License 1.1. You may not use this file
+ * except in compliance with the Zeebe Community License 1.1.
+ */
+package io.camunda.zeebe.engine.state.appliers;
+
+import io.camunda.zeebe.engine.processing.deployment.model.element.ExecutableFlowNode;
+import io.camunda.zeebe.engine.state.TypedEventApplier;
+import io.camunda.zeebe.engine.state.immutable.ProcessState;
+import io.camunda.zeebe.engine.state.instance.ElementInstance;
+import io.camunda.zeebe.engine.state.mutable.MutableElementInstanceState;
+import io.camunda.zeebe.engine.state.mutable.MutableProcessState;
+import io.camunda.zeebe.protocol.impl.record.value.processinstance.ProcessInstanceCreationRecord;
+import io.camunda.zeebe.protocol.record.intent.ProcessInstanceCreationIntent;
+import io.camunda.zeebe.protocol.record.value.BpmnElementType;
+import java.util.List;
+import org.agrona.DirectBuffer;
+
+final class ProcessInstanceCreationCreatedApplier
+    implements TypedEventApplier<ProcessInstanceCreationIntent, ProcessInstanceCreationRecord> {
+
+  private final ProcessState processState;
+  private final MutableElementInstanceState elementInstanceState;
+
+  public ProcessInstanceCreationCreatedApplier(
+      final MutableProcessState processState,
+      final MutableElementInstanceState elementInstanceState) {
+    this.processState = processState;
+    this.elementInstanceState = elementInstanceState;
+  }
+
+  @Override
+  public void applyState(final long key, final ProcessInstanceCreationRecord value) {
+    if (value.hasStartInstructions()) {
```

Comment:
🔧 nitpick: you could consider an early return here by inverting this if statement.